modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/LogicalScanConverterRule.java

```diff
@@ -125,7 +132,13 @@ public abstract class LogicalScanConverterRule<T extends ProjectableFilterableTa
                 RelOptCluster cluster = rel.getCluster();
                 IgniteTable table = rel.getTable().unwrap(IgniteTable.class);
 
-                RelDistribution distribution = table.distribution();
+                RelDistribution distribution;
+                QueryProperties qryProps = planner.getContext().unwrap(QueryProperties.class);
```

Comment:
QueryProperties is an ignite-core entity. It's not a good idea to use such an entities for rules, to keep rules transferable between ignite2 and ignite3. I think it's better to use some field in PlanningContext (or perhaps in BaseQueryContext to use it for MappingContext too)